ddtrace/internal/remoteconfig/products/apm_tracing.py

```diff
@@ -0,0 +1,82 @@
+import typing as t
+
+from ddtrace import config
+from ddtrace.internal.core.event_hub import dispatch
+from ddtrace.internal.core.event_hub import on
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.remoteconfig import Payload
+from ddtrace.internal.remoteconfig._connectors import PublisherSubscriberConnector
+from ddtrace.internal.remoteconfig._publishers import RemoteConfigPublisher
+from ddtrace.internal.remoteconfig._pubsub import PubSub
+from ddtrace.internal.remoteconfig._subscribers import RemoteConfigSubscriber
+
+
+requires = ["remote-configuration"]
+
+
+log = get_logger(__name__)
+
+
+def _rc_callback(payloads: t.Sequence[Payload]) -> None:
+    for payload in payloads:
+        if payload.metadata is None or (content := payload.content) is None:
+            continue
+
+        if (service_target := t.cast(t.Optional[dict], content.get("service_target"))) is not None:
+            if (service := t.cast(str, service_target.get("service"))) is not None and service != config.service:
+                continue
+
+            if (env := t.cast(str, service_target.get("env"))) is not None and env != config.env:
+                continue
+
+        if (lib_config := t.cast(dict, content.get("lib_config"))) is not None:
+            dispatch("apm-tracing.rc", (lib_config, config))
```

Comment:
@mabdinur  I understand this makes some things easier to test, but we cannot pass a specific config object because this a generic, cross-product event that we are emitting. Passing the global config (which currently is mostly tracer-oriented) is not relevant to, e.g., DI. I think we need to find a different approach 🤔